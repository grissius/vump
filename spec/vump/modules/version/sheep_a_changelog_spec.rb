require 'vump/modules/version/sheep_a_changelog'

mod = Vump::SheepAChangelog.new('', date: Time.new(2019, 0o2, 20))

RSpec.describe Vump::SheepAChangelog do
  it 'filename' do
    expect(mod.filename).to eql('CHANGELOG.md')
  end

  it 'select' do
    changelog = File.read(File.expand_path("examples/#{mod.filename}", __dir__))
    expect(mod.select(changelog)).to eql('0.1.0')
  end

  it 'compose' do
    changelog = File.read(File.expand_path("examples/#{mod.filename}", __dir__))
    expect(mod.compose(changelog, '1.1.2')).to match_snapshot
  end
end
