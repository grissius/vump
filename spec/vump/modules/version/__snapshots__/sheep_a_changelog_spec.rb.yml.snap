---
"./spec/vump/modules/version/sheep_a_changelog_spec.rb[1:3]": |
  # Changelog
  All notable changes to this project will be documented in this file.

  The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
  and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).

  ## [Unreleased]

  ## [1.1.2] - 2019-02-20

  ## [0.1.0] - 2019-01-14
  ### Added
  - Project boilerplate
  - Parser
  - Readme

  [Unreleased]: https://github.com/AckeeCZ/desmond/compare/v1.1.2...HEAD
  [1.1.2]: https://github.com/AckeeCZ/desmond/compare/v0.1.0...v1.1.2
  [0.1.0]: https://github.com/AckeeCZ/desmond/compare/c0b1f8c...v0.1.0
